### ISSUE TITLE
New Regions for describing between Tick Marks

### ISSUE THREAD

**Issue by terracoda** [phetsims/ratio-and-proportion#198]
@zepumph, @Matthew-Moore240 and I discussed some minor improvements to the describing the small steps between the tick marks. 

**Summary of Changes:** 
{{just below}} -> {{almost on}} 
{{below}} -> {{just over half-way past}}
{{above}} -> {{almost half-way past}} 
{{just above}} -> {{around}} 

**Example Object Responses for stepping between Tick Marks 2 and 3 (bolded steps are default step-size)**
Small & Large steps | Tick Marks checked | Numbered Tick Marks checked
-- | -- | --
**=10** | **{{on}} {{third}} tick mark** | **{{on}} tick mark {{3}}**
8-9 | {{almost on}} {{third}} tick mark | {{almost on}} tick mark {{3}}
6-7 | {{just over half-way past}} {{third}} tick mark | {{just over half-way past}} tick mark {{3}}
**=5** | **{{half-way past}} {{second}} tick mark** | **{{half-way past}} tick mark {{2}}**
3-4 | {{almost half-way past}} {{second}} tick mark | {{almost half-way past}} tick mark {{2}}
1-2 | {{around}} {{second}} tick mark | {{around}} tick mark {{2}}
**= 1** | **{{on}} {{second}} tick mark** | **{{on}} tick mark {{2}}**

**Comment by zepumph** [comment]
- [ ] @terracoda mentioned that we may still need to tweak the 0-1 range, because "almost half-way past tick mark 0" may sound strange.

**Comment by zepumph** [comment]
Implemented above. I'm a little confused by "{{just over half-way past}} {{third}} tick mark". My brain doesn't easily meld the "past" with the "third" when in between 2-3. That said, I don't feel strongly, and totally trust you. I thought I'd mention my perspective.

Please review.

**Comment by terracoda** [comment]
@zepumph, I agree, I was wondering if we could safely use "around" instead the "just over" phrasing.

Also, when I pass half-way past tick mark zero, the next step says "just over half-way past tick mark 1". The "1" needs to be "0" until "almost on tick mark 1."

I'm going to re-look at the whole scale for a minute to see if there should be any other changes.

**Comment by terracoda** [comment]
@zepumph said "I'm a little confused by "{{just over half-way past}} {{third}} tick mark". My brain doesn't easily meld the "past" with the "third" when in between 2-3. "

That was a typo, it should have been the previous number, "{{just over half-way past}} {{second}} tick mark"

But, still, the description is a little long and confusing.

I'm working on it.

**Comment by terracoda** [comment]
@zepumph, can you do this instead:

Small & Large steps | Tick Marks checked | Numbered Tick Marks checked
-- | -- | --
=10 | {{on}} {{third}} tick mark | {{on}} tick mark {{3}}
8-9 | {{almost on}} {{third}} tick mark | {{almost on}} tick mark {{3}}
6-7 | {{around}} {{half-way past}} {{second}} tick mark |{{around}} tick mark {{2.5}}
=5 | {{half-way past}} {{second}} tick mark | {{on}} tick mark {{2.5}}
3-4 | {{almost half-way past}} {{second}} tick mark | {{almost}} tick mark {{2.5}}
1-2 | {{around}} {{second}} tick mark | {{around}} tick mark {{2}}
= 1 | {{on}} {{second}} tick mark | {{on}} tick mark {{2}}

**Comment by terracoda** [comment]
@zepumph and @Matthew-Moore240, **I would like the above "in-between regions" implemented, and I want to post that I would be open to simplifying further if need be:**

Small & Large steps | Tick Marks checked | Numbered Tick Marks checked
-- | -- | --
=10 | {{on}} {{third}} tick mark | {{on}} tick mark {{3}}
8-9 | {{around}} {{third}} tick mark | {{around}} tick mark {{3}}
6-7 | {{around}} {{half-way past}} {{second}} tick mark |{{around}} tick mark {{2.5}}
=5 | {{half-way past}} {{second}} tick mark | {{on}} tick mark {{2.5}}
3-4 | {{around half-way past}} {{second}} tick mark | {{around}} tick mark {{2.5}}
1-2 | {{around}} {{second}} tick mark | {{around}} tick mark {{2}}
= 1 | {{on}} {{second}} tick mark | {{on}} tick mark {{2}}

**Comment by zepumph** [comment]
I like both! Sorry I can't be of more help. Let me know what you would like implemented.

**Comment by Matthew-Moore240** [comment]
I like the newest one @terracoda posted. I like some of the shorter responses.

**Comment by terracoda** [comment]
@zepumph, can you please implement the the responses in https://github.com/phetsims/ratio-and-proportion/issues/198#issuecomment-710029471

Thanks :-)

**Comment by zepumph** [comment]
Implemented above. Please review

**Comment by terracoda** [comment]
@zepumph, sounding good, but I found an odd string in the numbered tick mark experience.
I heard, "half-way past tick mark 7.5".
This should be "on tick mark 7.5" - so beautifully short :-)
The same problem in the Both Hands situation.
I hear, "left half-way past tick mark 7.5"

Since we are going with the half decimal, we need to remove "half-way past" string from the numbered tick marks responses.

**Comment by terracoda** [comment]
Here are some screen shots for help
Both Hands
![Screen Shot 2020-10-20 at 2 19 43 PM](https://user-images.githubusercontent.com/1902116/96618376-70898600-12df-11eb-939c-62e02a7cf257.png)

A single hand slider
![Screen Shot 2020-10-20 at 2 20 07 PM](https://user-images.githubusercontent.com/1902116/96618416-7bdcb180-12df-11eb-987b-f003ef388385.png)

**Comment by terracoda** [comment]
Found some more in the around half-way region
![Screen Shot 2020-10-20 at 2 20 07 PM](https://user-images.githubusercontent.com/1902116/96618650-c827f180-12df-11eb-9ae9-a872a1c90d60.png)

![Screen Shot 2020-10-20 at 2 23 59 PM](https://user-images.githubusercontent.com/1902116/96618772-edb4fb00-12df-11eb-95d2-722255681679.png)

**Comment by zepumph** [comment]
Arg!! I updated the "on X.5" description above.


>Found some more in the around half-way region

I didn't know what you meant by this. Is there another bug besides what I just fixed?

**Comment by terracoda** [comment]
and in "around half-way past" should just be "around X.5"
![Screen Shot 2020-10-21 at 12 43 42 PM](https://user-images.githubusercontent.com/1902116/96740268-3da4c800-139b-11eb-9011-14b78dfa6866.png)

**Comment by zepumph** [comment]
Ok, I understand the problem much better now. Thanks for going over that with me. I ended up rewriting the regions for this, because they were so dependent on the current tick mark view. They should be good to go now.

**Comment by terracoda** [comment]
Some issues for distance between tick marks. This might need to go in a new issue? But I am working on something else, so just plopping screen shots here for the record.
"0 and a half tick marks apart" should be
"half a tick mark apart"
![Screen Shot 2020-10-30 at 11 55 18 AM](https://user-images.githubusercontent.com/1902116/97717038-49814000-1aa7-11eb-9848-8322e9c61b59.png)

"1 tick marks" should be
"1 tick mark"
![Screen Shot 2020-10-30 at 11 57 33 AM](https://user-images.githubusercontent.com/1902116/97717426-b98fc600-1aa7-11eb-8f45-15b13f4e08d0.png)

**Comment by terracoda** [comment]
We need to adjust the strings when the tick mark value is less than 1 or maybe around 1/2?

- hands 0 and a half tick marks apart from each other, left half-way past tick mark 0.5, right on first tick mark
should be:
- hands, a half tick mark apart from each other, left half-way past tick mark 0, right on first tick mark

![Screen Shot 2020-11-03 at 1 24 13 PM](https://user-images.githubusercontent.com/1902116/98016084-1e189100-1dd8-11eb-92b2-f47029ee49c2.png)

**Comment by terracoda** [comment]
There is some funny business that we need to work on below the first tick mark.

**Comment by zepumph** [comment]
This is good enough for interviews tomorrow. We will pick this back up after them.

**Comment by terracoda** [comment]
@zepumph, I think my comments in https://github.com/phetsims/ratio-and-proportion/issues/198#issuecomment-719587503 are muddying the waters for this issue. The Both hands situation is more complicated because currently, in that view, the design is describing distance between hands in tick marks apart.

I'll create a new issue to address that issue. I'm thinking we should stick with qualitative distance across the different views and only go quantitative on the actual tick marks. I think this will simplify the design a great deal.

With that in mind there are a couple of edge cases for between tick marks when we are below first tick mark or below tick mark 1.

Here's are the new strings to address regions between the tick marks when less than 1.
**Please note:**
- The table represents small steps (0.1) using the shift key. The rows with bold text are the default step size of (0.5).
- The italicized text are special cases that are different from the general pattern that are present above 1.
- Please notice slight difference for the plain Tick Marks between:
    - {{around half-way past}} {{first}} tick mark (i.e. around tick mark 1.5) and 
    - {{around}} {{half-way to}} {{first}} tick mark (i.e., around tick mark 0.5)

- I went with "near zero" rather than "around zero" because it might stand out a bit more, but "around zero" might work just as well.
- I went with "at zero" and "near zero" in both plain and numbered tick marks rather than having different strings. Zero is zero and zero creates a fuzzy situation. I am hoping that using the same strings will simplify the fuzzy situations.

Small steps | Tick Marks checked | Numbered Tick Marks checked
-- | -- | --
**=1** | **{{on}} {{first}} tick mark** | **{{on}} tick mark {{1}}**
8-9 | {{around}} {{first}} tick mark | {{around}} tick mark {{1}}
6-7 | _{{around}} {{half-way to}} {{first}} tick mark_ | {{around}} tick mark {{0.5}}
**=5** | **_{{half-way to}} {{first}} tick mark_** | **tick mark {{0.5}}**
3-4 | _{{around}} {{half-way to}} {{first}} tick mark_ | {{around}} tick mark {{0.5}}
1-2 | _{{near}} zero_ | _{{near}} {{zero}}_
= 0 | **_{{at}} zero_** | **_{{at}} zero_**

**Comment by terracoda** [comment]
I should also say that for the values above first tick mark (above tick mark 1), the strings are correctly implemented.

**Comment by terracoda** [comment]
Let me know if the special cases for less than 1 are problematic for implementation.

**Comment by zepumph** [comment]
In design meeting today, @terracoda mentioned that if it was much harder in the code to spell out "zero" when numbered check marks are checked, then don't worry about it.

She also updated the design doc table for "{{at}} zero" and "{{near}} {{zero}}" for the NUMBERED case, where if it is easier to be consistent with the rest of the description for the numbered case, then that is ok. Like:

`{{on}} tick mark {{zero}}`


`{{near}} tick mark {{zero}}`

"near" is only for zero.

**Comment by terracoda** [comment]
@zepumph, issue #244 will address the complicated phrases underlined above that describe "tick marks apart". Those phrases will be replaced with qualitative distance phrases which are more straight forward.

**Comment by zepumph** [comment]
After discussion with @terracoda right now, we found a typo in https://github.com/phetsims/ratio-and-proportion/issues/198#issuecomment-728344102, it should say "{{on}}" tick mark .5.

**Comment by zepumph** [comment]
The zero-case has been implemented above. I'm not happy with the complexity it added to this function. I feel like this is the most complicated describer function I have ever written. That complexity is simply because the description region cases are complicated. I am alright with incurring this cost, because @terracoda and I have discussed here the value of this zero-specific edge case. That said, it is worth re-iterating for this issue that there is no good established pattern for handling such knit-picky description regions. @terracoda please review.

**Comment by terracoda** [comment]
These sound good. Marking for meeting to discuss issue @zepumph referred to.

**Comment by terracoda** [comment]
Thanks great work.
